lib/kubernetes-deploy/kubernetes_resource/horizontal_pod_autoscaler.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class HorizontalPodAutoscaler < KubernetesResource
+    PRUNABLE = true
+    TIMEOUT = 30.seconds
+
+    def deploy_succeeded?
+      conditions = @instance_data.dig("status", "conditions") || []
+      able_to_scale = conditions.detect { |c| c["type"] == "AbleToScale" } || {}
+      able_to_scale["status"] == "True"
+    end
+
+    def deploy_failed?
+      !exists?
```

Comment:
I think this should be checking the `ScalingActive` condition too (but should the deploy should succeed when that's false because `ScalingDisabled`).